### DISCUSSION TITLE
Default cluster has no server defined when enabled SSO

### DISCUSSION THREAD

**Discussion by wcorbett36** [argoproj/argo-workflows#9211] [Q&A]
Hi all,  

My group is moving argo-wf from poc to operational stage and I'm attempting to setup SSO auth-mode as part of the process.  

When I enable auth-mode sso my pod goes to a crash state and logs:

Error: invalid configuration: default cluster has no server defined

I'm unable to find good documentation and asking here for pointers.  

Here is my relevant argo-server arguments:  

`    spec:
      containers:
      - args:
        - server
        - --configmap=argo-workflows-argowfopensource-workflow-controller-configmap
        - --auth-mode=sso
        - --verbose
        - --insecure-skip-tls-verify `

which throws error:  

Error: invalid configuration: default cluster has no server defined

If I change my auth-mode back to server it will launch successfully. 

thanks

**Comment by fhochleitner** [comment]
If you have found anything out yet I'd love to see you share your solution. currently fighting with exactly the same issue

**Comment by MichalM07** [comment]
@fhochleitner @wcorbett36 
use  this in the argoworkflow configmap 
sso:
  insecureSkipVerify: true 
  
not 
   - --insecure-skip-tls-verify

**Reply by fhochleitner** [reply]
thank you @MichalM07 for your reply. Will hopefully manage to try it out tomorrow.

**Reply by fhochleitner** [reply]
update: worked. unfortunately I realized that I can't directly use openshift as oidc provider for workflows. or I did not manage to do so. trying the dex route now.

**Reply by MichalM07** [reply]
If you are using already ArgoCd under LDAP you basically need to copy the url that you are redirected to when clicking the button to log via ldap.  Like in my case its:

https://SOME_IP:SOME_PORT/api/dex   

And this is my issuer
